scheduler/core/programprovider/gpp/gppprogramprovider.py

```diff
@@ -929,6 +812,66 @@ def parse_target(self, data: dict, targ_type: str) -> Target:
             msg = f'Illegal target type.'
             raise ValueError(msg)
 
+    def parse_observing_mode(self, data: dict) -> Tuple[FrozenSet[Resource], Wavelength, ObservationMode]:
+
+        def find_filter(filter_input: str, filter_dict: Mapping[str, float]) -> Optional[str]:
+            return next(filter(lambda f: f in filter_input, filter_dict), None)
+
+        instrument = GppProgramProvider._gpp_inst_to_ocs[data['instrument']]
+        mode = data['mode']
+
+        instrument_config = data.get('gmosNorthLongSlit') or data.get('gmosSouthLongSlit')
+
+        fpu = None
+        if instrument in GppProgramProvider.FPU_FOR_INSTRUMENT:
+            if GppProgramProvider._FPUKeys.CUSTOM in instrument_config.keys():
+                # This will assign the MDF name to the FPU
+                fpu = instrument_config[GppProgramProvider._FPUKeys.CUSTOM]
+            elif GppProgramProvider.FPU_FOR_INSTRUMENT[instrument] in instrument_config.keys():
+                fpu = instrument_config[GppProgramProvider.FPU_FOR_INSTRUMENT[instrument]]
+
```

Comment:
Maybe this is cleaner:
 ```
   fpu_key = GppProgramProvider.FPU_FOR_INSTRUMENT.get(instrument)
   fpu = instrument_config.get(GppProgramProvider._FPUKeys.CUSTOM) or instrument_config.get(fpu_key)
```

I think this does the same as the nested if statements